src/python/pants/engine/internals/mapper.py

```diff
@@ -47,6 +47,7 @@ def parse(
         parser: Parser,
         extra_symbols: BuildFilePreludeSymbols,
         env_vars: EnvironmentVars,
+        pants_hashes: Mapping[tuple[str, ...], str],
```

Comment:
I didn't bother creating a dedicated type for this, as it's not part of any rule API.. but did consider doing it anyway for aesthetic reasons. So if there's any preferences in that direction... 🤷🏽